Integrate enable contacts modal in invite friends

buoyad opened the pull request:
Ticket was for team invite by contacts, but I hit problems there. I made a ticket to fix that up.

~This adds `useEnableContactsPopup` that takes whether the user has denied contact permissions and an `onClose` and returns the contact modal element if we should be showing it.~

The reason it's a hook and not just `{noAccess && <EnableContacts onClose={navUp} />}` is because the unmount from `navUp` comes too late, so the popup stays open for too long. The hook dismisses the popup immediately on hitting close.

**Edit: this behavior ^ was all added to the `EnableContactsPopup` component.**

Also fixes the underlay color of `MobilePopup` in dark mode.

cc @keybase/y2ksquad

jakob223 commented on shared/teams/common/use-contacts.native.tsx:
```diff
@@ -3,7 +3,7 @@ import * as Container from '../../util/container'
 import * as SettingsConstants from '../../constants/settings'
 import * as SettingsGen from '../../actions/settings-gen'
-import {Contact} from './index.native'
+import {Contact} from '../invite-by-contact/index.native'
```
maybe `Contact` should live here?

jakob223 replied:
or in `Types` perhaps

jakob223 commented on shared/teams/common/enable-contacts.tsx:
```diff
@@ -28,9 +28,24 @@ const EnableContacts = ({onClose}: {onClose: () => void}) => {
   )
 }
 
+export const useEnableContactsPopup = (noAccess: boolean, onClose: () => void) => {
```
As we discussed, let's switch this to be a component instead of a hook.

Also, can you add a js doc comment to it explaining what it does, I think its props are non-obvious.

buoyad commented:
@jakob223 thanks, addressed feedback. I didn't specifically mention  the close behavior in the jsdoc, that seems like internal behavior users don't have to worry about.
